### DISCUSSION TITLE
How to configure MibS in  PAO?

### DISCUSSION THREAD

**Discussion by yhy0258** [coin-or/MibS#105] [Q&A] [answered]
Hi,

I install the MibS using coinbrew and Msy2 on Win11. I add coinbrew\dist\bin to the system path and can run mibs.exe successfully.
However, when I use mibs by PAO in an anaconda environment, 
```
opt = Solver('pao.pyomo.MIBS')`
results = opt.solve(M)
# The final solution is loaded into the model
print(M.x.value)
print(M.y.value)
print(results.solver.termination_condition)
print(results.check_optimal_termination())
```
the output is 
```
0
0
TerminationCondition.unknown
False
```
I find that MibS seem to be not invoked.
Then I open an anaconda prompt and run mibs.exe. It outputs an error “Entry Point Not Found Dynamic Link Library”.
How to configure MibS in  PAO?

Thank you.

**Comment by tkralphs** [comment, marked as answer]
I can only guess as to what's going on and you would probably be better off asking this question in the forums of PAO than here. But I suspect that the issue is that you have built with g++ using shared. Anaconda on Windows probably cannot open the shared libraries produced by g++. It may work to re-build with `--enable-static --disable-shared`. If not, I'm pretty sure that building with the visual studio compiler instead of g++ will work. Let me know if I can be of more assistance. Also, if you get it to work, it would be helpful if you could come back here and let us know the solution.

**Reply by yhy0258** [reply]
Thank you for your help.  I have removed the anaconda distribution and use python directly, which is OK.

**Comment by abosch7** [comment]
Hello, I have the same problem as you. I'm using Win11.

To install the MibS I used the "MibS-releases.1.2.0-i686-w64-mingw32-dbg.zip](https://github.com/coin-or/MibS/releases/download/releases%2F1.2.0/MibS-releases.1.2.0-i686-w64-mingw32-dbg.zip" and then, introduced as a Enviormental Variable in the "path" section. 

For PCCG it worked good, but for MibS it return me the same result as yhy0258.

I dont know what can I do, any help will be appreciated, 
Thank you.

**Reply by tkralphs** [reply]
The issue is probably https://github.com/or-fusion/pao/issues/98. See also https://github.com/or-fusion/pao/pull/101.
